software/grapherrific.livecodescript

```diff
@@ -0,0 +1,1477 @@
+   -- exits repeat as soon as undo enabled
+   repeat for each key tDoc in pMain
+      put "false" into pShort[tDoc][0]["undo"]
+      put "false" into pShort[tDoc][0]["redo"]
+      put the keys of pMain[tDoc] into tRecords
+      sort tRecords numeric descending
+      repeat for each line tRec in tRecords
+         if "turn" is among the keys of pMain[tDoc][tRec] then
+            if pMain[tDoc][tRec]["act"] contains "do" then next repeat
+            if clearNegPairs(pMain[tDoc][tRec]["turn"]) is empty then 
+               --               set the disabled of button "redoButt" stack "primary" to false
+               put "true" into pShort[tDoc][0]["redo"]
+            else
+               --               set the disabled of button "undoButt" stack "primary" to false
+               put "true" into pShort[tDoc][0]["undo"]
+               exit repeat
+            end if
+         end if
+      end repeat
+   end repeat 
+   -- metadata
+   repeat for each key tDoc in pMain
+      put pMain[tDoc][0]["pointerTags"] into pShort[tDoc][0]["pointerTags"]
+   end repeat
+end adjList
+
+command topoSort @pShort,@pTopo,@pNoViz,@pNotes
+   put "command topoSort" into xX["write"];    devLog xX
+   put empty into pTopo -- will contain the sorted nodes
+   put empty into tPlanA -- scope nodes outside of flow
+   put empty into tPlanB -- unscoped nodes at beginning of flow
+   put empty into pNotes -- treats nodes linked in by bond as "footnotes"
+   ---------------- list of seed nodes ----------------------
+   repeat for each key tDoc in pShort
+      put the keys of pShort[tDoc] into pNoViz[tDoc]
+      replace cr with "," in pNoViz[tDoc]
+      repeat for each key tRec in pShort[tDoc]
+         if pShort[tDoc][tRec]["type"] is "state" or pShort[tDoc][tRec]["type"] is "change" then
+            put the keys of pShort[tDoc][tRec] into tTheKeys
+            put false into tPrev
+            put false into tNext
+            put false into tRead
+            put false into tBulk
+            put false into tSkim
+            if "prev" is among the lines of tTheKeys then put true into tPrev
+            if "next" is among the lines of tTheKeys then put true into tNext
+            if "read" is among the lines of tTheKeys then put true into tRead
+            if "bulk" is among the lines of tTheKeys then put true into tBulk
+            if "skim" is among the lines of tTheKeys then put true into tSkim
+            if tRead and tBulk and not tNext and not tPrev then put tRec & "," after tPlanA
+            if tNext and not tPrev and not tSkim and not tBulk then put tRec & "," after tPlanB
+         end if
+      end repeat -- tRec
+      put tPlanA after tNextList
+      put tPlanB after tNextList
+      get kahnSort(pShort,pTopo,pNoViz,pNotes,tDoc,tNextList) 
+   end repeat -- tDoc
+end topoSort
+
+function kahnSort @pShort,@pTopo,@pNoViz,@pNotes,pDoc,pSeedList
+   put "function kahnSort" into xX["write"];    devLog xX
```

Comment:
@matthewmaier As you are using `wholeMatches = true` in all your `itemOffset` calls in this function I would record the current wholeMatches value at the start and set it to true then reset it again at the end.